### PR TITLE
K8SPG-258: Add PerconaPGCluster

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Review by pooknull — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/039e4afcd0824f4c0369528f6245ac36cc5feddd
image: `perconalab/percona-postgres-operator:PR-303-039e4afc`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/a6ebb528acd59d81a35631f282dd06065677b920
image: `perconalab/percona-postgres-operator:PR-303-a6ebb528`

File: e2e-tests/vars.sh [hunk 1]
```diff
@@ -10,7 +10,16 @@ export TEMP_DIR=$(mktemp -d)
...
 export VERSION=${VERSION:-$(echo "${GIT_BRANCH}" | sed -e 's^/^-^g; s^[.]^-^g;' | tr '[:upper:]' '[:lower:]')}
 
-export IMAGE=${IMAGE:-"perconalab/percona-postgresql-operator:${VERSION}"}
+export IMAGE_BASE=${IMAGE_BASE:-"perconalab/percona-postgresql-operator"}
+export IMAGE=${IMAGE:-"${IMAGE_BASE}:${VERSION}"}
+export PG_VER="${PG_VER:-14}"
+export IMAGE_PGBOUNCER=${IMAGE_PGBOUNCER:-"${IMAGE_BASE}:main-ppg$PG_VER-pgbouncer"}
+export IMAGE_POSTGRESQL=${IMAGE_POSTGRESQL:-"${IMAGE_BASE}:main-ppg$PG_VER-postgres"}
+export IMAGE_BACKREST=${IMAGE_BACKREST:-"${IMAGE_BASE}:main-ppg$PG_VER-pgbackrest"}
+export IMAGE_BACKREST_REPO=${IMAGE_BACKREST_REPO:-"${IMAGE_BASE}:main-ppg$PG_VER-pgbackrest-repo"}
```
Comment (hors) [inline e2e-tests/vars.sh:19]: We do not have separate image for backrest repo

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/9d2b608a68c87ea1b780d631fc98e0aa5ee2debc
image: `perconalab/percona-postgres-operator:PR-303-9d2b608a`

File: cmd/postgres-operator/main.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ import (
 	"github.com/crunchydata/postgres-operator/internal/logging"
 	"github.com/crunchydata/postgres-operator/internal/upgradecheck"
 	"github.com/crunchydata/postgres-operator/internal/util"
+
+	percona "github.com/crunchydata/postgres-operator/percona/controllers"
```
Comment (hors) [inline cmd/postgres-operator/main.go:35]: I think we should not use` github.com/crunchydata`

**Review by hors — CHANGES_REQUESTED** [review]

File: deploy/cr.yaml [hunk 3]
```diff
@@ -144,14 +145,14 @@ spec:
 #            matchLabels:
 #              postgres-operator.crunchydata.com/role: pgbouncer
 #
-#      containers:
+#      sidecars:
```
[the remark this answers is not in the document]
Comment (egegunes) [reply deploy/cr.yaml:148]: Sidecars are bound to a feature gate by default. You need to run operator with `PGO_FEATURE_GATES=InstanceSidecars=true` env variable to enable

File: cmd/postgres-operator/main.go (hunk 2 above)
Comment (egegunes) [reply cmd/postgres-operator/main.go]: https://github.com/percona/percona-postgresql-operator/pull/303/commits/93e3e6a901eae6c600f0d2d865b4080f2ebdb122

**Comment by egegunes** [comment]
@cap1984 

* Fixed proxy crash in https://github.com/percona/percona-postgresql-operator/pull/303/commits/1f07a09004afd4ed627bb3b5c38178c4539cf17d
* Now our controller pass annotations to PostgresCluster: https://github.com/percona/percona-postgresql-operator/pull/303/commits/99192e4c6c591736b5b7514a5360e4fed4dfa27e

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/93e3e6a901eae6c600f0d2d865b4080f2ebdb122
image: `perconalab/percona-postgres-operator:PR-303-93e3e6a9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/97354469bb5ea0a4f89d9723eafc799a24e55044
image: `perconalab/percona-postgres-operator:PR-303-97354469`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/2fcfcc46/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/2fcfcc466ca66481ea0f31f0cd82c326df995bbf
image: `perconalab/percona-postgres-operator:PR-303-2fcfcc46`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/fda1a019/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/fda1a019a07e45bd194ab78cc3b3f87b5bb62d02
image: `perconalab/percona-postgres-operator:PR-303-fda1a019`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/a38c535d93d3b22c4b9e3010b1177413dc37fa53
image: `perconalab/percona-postgres-operator:PR-303-a38c535d`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/00babb6b/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/00babb6bd089679bd884d73ff08320fdf753a265
image: `perconalab/percona-postgres-operator:PR-303-00babb6b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/3e7d0263/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/3e7d0263302ade276256228e89fdae714d1c49d2
image: `perconalab/percona-postgres-operator:PR-303-3e7d0263`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/74017449/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/7401744900aa4a13c2e33c4829eb2731d9860b89
image: `perconalab/percona-postgres-operator:PR-303-74017449`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/1c9aab97/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/1c9aab979b996dc1bacce854a26e5d1825b36fed
image: `perconalab/percona-postgresql-operator:PR-303-1c9aab97`

File: e2e-tests/vars.sh (hunk 1 above)
Comment (cap1984) [reply e2e-tests/vars.sh]: done

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/e3dbdd0f/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/e3dbdd0ffd51a7f0bbed02bdfc36e19445363163
image: `perconalab/percona-postgresql-operator:PR-303-e3dbdd0f`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/96a185c6/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/96a185c6618858a6999ac6218ece683e57262c1c
image: `perconalab/percona-postgresql-operator:PR-303-96a185c6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator-v2/PR-303/428712ac/init-deploy.log) |
| We run 1 out of 1|

commit: https://github.com/percona/percona-postgresql-operator/pull/303/commits/428712acf85f9a21187698d24643edfaf96ce64d
image: `perconalab/percona-postgresql-operator:PR-303-428712ac`

**Review by hors — APPROVED** [review]

**Pull request merged**